Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -456,7 +456,50 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
         ],
         handledNodes: [argument.id]
       )
-      return .visitChildren
+    }
+    if node.hasError,
+       let unexpectedBeforeAtSignTokens = node.unexpectedBeforeAtSignToken,
+       let unexpectedBeforeAtSign = unexpectedBeforeAtSignTokens.lastToken(viewMode: .sourceAccurate) {
+      let tokensToDiscard = unexpectedBeforeAtSignTokens.compactMap { $0.as(TokenSyntax.self) }
+      if unexpectedBeforeAtSign.tokenKind == .atSign, !unexpectedBeforeAtSign.trailingTrivia.isEmpty {
+        addDiagnostic(
+          unexpectedBeforeAtSign,
+          StaticParserError.invalidWhitespaceBetweenAttributeAtSignAndIdentifier,
+          fixIts: [
+            FixIt(
+              message: StaticParserFixIt.removeExtraneousWhitespace,
+              changes: [
+                .makeMissing(tokensToDiscard, transferTrivia: false),
+                .makePresent(node.atSignToken),
+              ]
+            )
+          ],
+          handledNodes: [
+            node.id,
+            unexpectedBeforeAtSignTokens.id,
+            node.atSignToken.id
+          ]
+        )
+      } else if unexpectedBeforeAtSign.tokenKind == .identifier(unexpectedBeforeAtSign.text) {
```

Comment:
This condition seems odd to me. Why would we have an identifier *before* the `@`?